packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart

```diff
@@ -462,12 +466,97 @@ class DebuggerController extends DisposableController
   void selectStackFrame(StackFrameAndSourcePosition frame) {
     _selectedStackFrame.value = frame;
 
+    if (frame?.frame != null) {
+      _variables.value = _variablesForFrame(frame.frame);
+    } else {
+      _variables.value = [];
+    }
+
     if (frame?.scriptRef != null) {
       showScriptLocation(
           ScriptLocation(frame.scriptRef, location: frame.position));
     }
   }
 
+  List<Variable> _variablesForFrame(Frame frame) {
+    final variables = frame.vars.map((v) => Variable.create(v)).toList();
+    variables.forEach(buildVariablesTree);
+    return variables;
+  }
+
+  Future<void> buildVariablesTree(Variable variable) async {
```

Comment:
can you add brief docs about what this method does? It doesn't return anything; it populates the children of a variable?

Reply:
done